pyClient/zeth/contracts.py

```diff
@@ -23,6 +23,9 @@
 
 Interface = Dict[str, Any]
 
+# Installing and setting the right version of the compiler to use
+install_solc_pragma(constants.SOLC_VERSION_PRAGMA)
```

Comment:
Just one suggestion: If I've understood the setup correctly, the compiler will be installed whenever the `contract` module is included.  We could either execute `python -m zeth.contracts` as part of `make setup`, or (maybe better) we could do the install only when oen of the `compile` functions in this module is called.  In that way, clients that do not deploy the contract can avoid the install altogether.

Reply:
> we could do the install only when the compile functions in this module is called.

This solution probably is the best, but there are two annoying things:

- since we have many different `compile_*` primitives, if we add the install instruction multiple time we will see many times this log message, that we are unable to make it silent:

``
solc, the solidity compiler commandline interface                                                                                            
Version: 0.5.16+commit.9c3226ce.Linux.g++   
``

- repetitive instruction: 

`install_solc_pragma(constants.SOLC_VERSION_PRAGMA)`

repeated many times.

Reply:
What about we add 

`install_solc_pragma(SOLC_VERSION_PRAGMA)` 

into `setup.py` script?

Reply:
Yeah, tbh either of those mechanisms seems fine.  Just a few comments.

> since we have many different compile_* primitives, if we add the install instruction multiple time we will see many times this log message, that we are unable to make it silent

This could be addressed if there is some kind of `get_installed_solc_versions`, so we can avoid unnecessary calls to `install_solc_pragma`.

But as part of the optimization work I switched to have just one contracat so one compile function (or maybe 2, including the token).  So practically, this message will only be seen once when deploying, and maybe once when the token is deployed.  (And I guess in general the token deployment will be separate from the zeth deployment).

> What about we add `install_solc_pragma(SOLC_VERSION_PRAGMA)` into setup.py script?

If it's difficult to do the above then installing from setup.py (or anywhere as part of `make setup`) is fine I think.

The main reason I'm focussing on making it on-demand is just that it's a shame to do the install on clients that do not require it (it installs dependencies and does a build on mac).  But this is not a huge issue, so not worth putting a lot of effort into at this stage.

I'm happy with whichever of these you prefer.

Reply:
Alright, to me make sense to separate the compiler installation from the call of compiling/deploy a contract and make it as part of the setup phase.

Adding this to the `make setup` seems working:

```
python -c "from solcx import install_solc_pragma; from zeth.constants import SOLC_VERSION_PRAGMA; install_solc_pragma(SOLC_VERSION_PRAGMA)"
```

I'll push it like these and then we can change it in the future if we find something better.